Library/Homebrew/cmd/missing.rb

```diff
@@ -6,15 +6,41 @@ module Homebrew
   def missing
     return unless HOMEBREW_CELLAR.exist?
 
-    ff = if ARGV.named.empty?
+    formulae = if ARGV.named.empty?
       Formula.installed
     else
       ARGV.resolved_formulae
     end
+    if ARGV.json == "v1"
+      missing = print_missing_json(formulae)
+    else
+      missing = print_missing(formulae)
+    end
+    Homebrew.failed = ARGV.resolved_formulae.any? && missing.any?
+  end
+
+  def print_missing(formulae)
+    verbose = ($stdout.tty? || ARGV.verbose?) && !ARGV.flag?("--quiet")
+    s = Set.new
+    missing = Diagnostic.missing_deps(formulae) do |name, m|
+      if verbose
+        print "#{name}: " if formulae.size > 1
+        puts "#{m.map(&:full_name) * " "}"
+      else
+        m.each { |f| s << f }
+      end
+    end
+    s.each { |f| puts f.full_name } unless verbose
```

Comment:
I find the intermixed parts that deal with verbose and non-verbose output a bit hard to read. I don't think it is necessary to populate a new set when the same information is also duplicated in the `missing` hash. It just needs to be slightly transformed for the non-verbose output.